homework-08/main.js

```diff
@@ -0,0 +1,142 @@
+    {productName: 'butter', productPrice:  25, productQuantity: 1},
+    {productName: 'milk', productPrice:  17, productQuantity: 3},
+    {productName: 'flour', productPrice:  12.75, productQuantity: 1},
+    {productName: 'sugar', productPrice:  35, productQuantity: 5},
+    {productName: 'salt', productPrice:  10, productQuantity: 4},
+    {productName: 'oil', productPrice:  45, productQuantity: 7},
+    {productName: 'vinegar', productPrice:  10, productQuantity: 1},
+    {productName: 'nuts', productPrice:  18, productQuantity: 6},
+]
+
+// Порахувати загальну вартість корзини та вивести суму у зрозумілому форматі
+const totalProductPrice = PRODUCT_ARRAY.reduce(
+    (accumulator, item) => {
+        return accumulator + item.productPrice * item.productQuantity;    
+    },
+0);
+
+console.log(`The total price of products is ${totalProductPrice}`);
+
+// Порахувати найменшу кількість продукту, який потрібно купити
+const result = PRODUCT_ARRAY.reduce((minValue, curValue) => 
+    curValue.productQuantity < minValue ? curValue.productQuantity : minValue
+, Infinity); 
+console.log(result)
+
+// Порахувати загальну кількість продуктів
+const totalProducAmount = PRODUCT_ARRAY.reduce(
+    (accumulator, item) => {
+        return accumulator + item.productQuantity;    
+    },
+0);
+console.log(totalProducAmount);
+
+// Знайти найдорожчий продукт
+expensiveProduct = PRODUCT_ARRAY.sort((a, b) => b.productPrice - a.productPrice)[0]
+console.log(`The most Expensive product in list is ${expensiveProduct.productName} costs ${expensiveProduct.productPrice}`)
+
+
+// Створити функцію, яка повинна додавати новий продукт в існуючий масив
+function isParamValid(param, allowNumeric=false)
+{
+    if ( param === '' || param === null || param === undefined )
+        return false;
+
+    if (allowNumeric && (isNaN(param) || +param < 1))
+        return false;
+
+    return true;
+}
+
+function addNewProduct(products) {
+    productName = prompt('Enter the product name');
+    productPrice = prompt('Enter price');
+    productQuantity = prompt('Enter Quantity');
+
+    if (isParamValid(productName) && isParamValid(productPrice, true) && isParamValid(productQuantity, true))
+        products.push( {productName, productPrice, productQuantity} )
+    else
+        console.log("Something wrong with one of params!")
+
+    return products;
+}    
+console.log(addNewProduct(PRODUCT_ARRAY));
+
+// Створити функцію, яка повинна видаляти конкретний продукт із існуючий масиву продуктів
+function deleteProduct(products) {
+    if (!products instanceof Array || products == [])
+        throw "Parameter is not array or empty"
+
+    let value = prompt('Enter product name to delete');
+    if (!isParamValid(value))
+        console.log("USer entered wrong value")
+
+    let indexOfProduct = products.findIndex(element => element.productName === value);
+    if(indexOfProduct < 0)
+        console.log("The entered product not found")
+

```

Comment:
Please, lint your code. A lot of this mistakes occur

```suggestion
    if(indexOfProduct < 0) {
        console.log("The entered product not found")
    }

```